dds_web/api/user.py

```diff
@@ -533,11 +519,23 @@ class Token(flask_restful.Resource):
     @basic_auth.login_required
     @logging_bind_request
     def get(self):
-        return flask.jsonify({"token": jwt_token(username=auth.current_user().username)})
+        args = flask.request.json
+        if args is None:
+            args = {}
 
+        data = token_schemas.TokenSchema().load(args)
+
+        return flask.jsonify(
+            {
+                "token": encrypted_jwt_token(
+                    username=auth.current_user().username, sensitive_content=None
+                )
+            }
+        )
 
-class EncryptedToken(flask_restful.Resource):
-    """Generates encrypted token for the user."""
+
+class RequestMail2fa(flask_restful.Resource):
```

Comment:
I would replace this class with the following

`class SecondFactor(flask_restful.Resource):
    """Take in and verify an authentication One-Time code entered by an authenticated user with basic credentials"""

    @auth.login_required
    def get(self):
        args = flask.request.json
        if args is None:
            args = {}

        token_schemas.TokenSchema().load(args)

        token_claims = dds_web.security.auth.decrypt_and_verify_token_signature(flask.request.headers["Authorization"].split()[1])

        return flask.jsonify(
            {
                "token": update_token_with_mfa(token_claims)
            }
        )`

Reply:
For the previous code, update_token_with_mfa should be defined in dds_web/security/tokens.py as 

```
def update_token_with_mfa(token_claims):
    expires_in = datetime.datetime.fromtimestamp(token_claims.get("exp")) - dds_web.utils.current_time()
    return encrypted_jwt_token(username=token_claims.get("sub"),
                               sensitive_content=None,
                               expires_in=expires_in,
                               additional_claims={"mfa_auth_time": dds_web.utils.current_time().timestamp()})

```

Reply:
Not sure I understand the row `expires_in = datetime.datetime.fromtimestamp(token_claims.get("exp")) - dds_web.utils.current_time()`.  What change does that make to the expiration time of the token?

Reply:
Nevermind I think I understand, it's because you add that time when you create the token in the other method.